src/mono/mono/mini/aot-compiler.c

```diff
@@ -12407,25 +12407,15 @@ emit_file_info (MonoAotCompile *acfg)
 	init_aot_file_info (acfg, info);
 
 	if (acfg->aot_opts.static_link) {
-		char symbol [MAX_SYMBOL_SIZE];
+		char prefix [MAX_SYMBOL_SIZE];
 
 		/*
 		 * Emit a global symbol which can be passed by an embedding app to
 		 * mono_aot_register_module (). The symbol points to a pointer to the file info
 		 * structure.
 		 */
-		sprintf (symbol, "%smono_aot_module_%s_info", acfg->user_symbol_prefix, acfg->image->assembly->aname.name);
-#ifdef TARGET_WASM
-		acfg->static_linking_symbol = g_strdup (mono_fixup_symbol_name(symbol));
-#else
-		/* Get rid of characters which cannot occur in symbols */
-		char *p = symbol;
-		for (p = symbol; *p; ++p) {
-			if (!(isalnum (*p) || *p == '_'))
-				*p = '_';
-		}
-		acfg->static_linking_symbol = g_strdup (symbol);
-#endif	
+		sprintf (prefix, "%smono_aot_module_", acfg->user_symbol_prefix);
+		acfg->static_linking_symbol = g_strdup (mono_fixup_symbol_name(prefix, acfg->image->assembly->aname.name, "_info"));
```

Comment:
Don't need the `g_strdup` if you update `mono_fixup_symbol_name` to always return freshly allocated memory.